app/models/spree/line_item_decorator.rb

```diff
@@ -0,0 +1,46 @@
+Spree::LineItem.class_eval do
+
+  after_commit :create_line_item_added_event_on_intercom, on: :create, if: :user_present?
+  after_commit :create_line_item_updated_event_on_intercom, on: :update, if: [:user_present?, :quantity_updated_to_non_zero_quantity?]
+  after_commit :create_line_item_removed_event_on_intercom, on: :destroy, if: :user_present?
+
+  private
+
+    def user_present?
+      order.user_id.present?
+    end
+
+    def create_line_item_added_event_on_intercom
```

Comment:
see if we can move these out in a concern as well, like I explained for the controllers above